technic/machines/register/recipes.lua

```diff
@@ -57,40 +57,38 @@ local function register_recipe(typename, data)
 	local recipe = {time = data.time, input = {}, output = data.output}
 	local index = get_recipe_index(data.input)
 	if not index then
-		print("[Technic] ignored registration of garbage recipe!")
+		minetest.log("warning", "[Technic] ignored registration of garbage recipe!")
 		return
 	end
 	for _, stack in ipairs(data.input) do
 		recipe.input[ItemStack(stack):get_name()] = ItemStack(stack):get_count()
 	end
 
 	technic.recipes[typename].recipes[index] = recipe
-	if have_ui and technic.recipes[typename].output_size == 1 then
-		unified_inventory.register_craft({
-			type = typename,
-			output = data.output,
-			items = data.input,
-			width = 0,
-		})
-	end
-	if (have_cg or have_i3) and technic.recipes[typename].output_size == 1 then
-		local result = data.output
-		if (type(result)=="table") then
-			result = result[1]
+	if data.hidden then return end
```

Comment:
Would be good if this return in the middle of function can be refactored to not be here.

Pretty sure this function should be split into two for readability, everything below this line would probably be better split into another local function.

Reply:
I don't know, seems unnecessary to split it, even for readability. I think I'll just leave it for now, it can be refactored later when group support is added.